lib/config/error-codes/throw.js

```diff
@@ -72,25 +84,84 @@ function throwError (domain, subdomain, error, ...placeholders) {
   throw getError(domain, subdomain, error, ...placeholders);
 }
 
+
 /**
- * Wrap getError and throw with the provided domain and subdomain.
+ * Construct and return the corresponding error
+ * @param  {string} errorSource - non KuzzleError from which the KuzzleError will derivate
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function getErrorFrom (errorSource, domain, subdomain, error, ...placeholders) {
 
- * The function have now the following signature:
+  const
+    errorName = `${domain}.${subdomain}.${error}`,
+    kuzzleError = _.get(errorCodes, `${domain}.subdomains.${subdomain}.errors.${error}`);
+
+  if (! kuzzleError) {
+    return getErrorFrom(
+      errorSource,
+      'internal',
+      'unexpected',
+      'unknown_error',
+      ...placeholders);
+  }
+
+  const code =
+    (errorCodes[domain].code >> 24) |
```

Comment:
Wrong. See https://github.com/kuzzleio/kuzzle/pull/1404

Also, it might be a good idea to DRYify this code calculation.